samtranslator/model/eventsources/push.py

```diff
@@ -600,6 +603,56 @@ def _add_swagger_integration(self, api, function):
                 editor.add_request_model_to_method(path=self.Path, method_name=self.Method,
                                                    request_model=self.RequestModel)
 
+        if self.RequestParameters:
+
+            default_value = {
+                'Required': False,
+                'Caching': False
+            }
+
+            parameters = []
+            for parameter in self.RequestParameters:
+
+                if isinstance(parameter, dict):
+
+                    parameter_name, parameter_value = next(iter(parameter.items()))
+
+                    if not re.match('method\.request\.(query|path|header)', parameter_name):
+                        raise InvalidEventException(
+                            self.relative_id,
+                            "Invalid value for 'RequestParameters' property")
+
+                    if not isinstance(parameter_value, dict) or not all(key in RequestParameterProperties
+                                                                        for key in parameter_value.keys()):
+                        raise InvalidEventException(
+                            self.relative_id,
+                            "Invalid value for 'RequestParameters' property")
+
+                    settings = default_value.copy()
+                    settings.update(parameter_value)
+                    settings.update({'Name': parameter_name})
+
+                    parameters.append(settings)
+
+                elif not isinstance(parameter, string_types):
+                    raise InvalidEventException(
+                        self.relative_id,
+                        "Invalid value for 'RequestParameters' property")
+
+                else:
```

Comment:
Can we swap this and the previous condition around, so check if it's a string, and then the final else (no elif) will be the case that it's not an object or a string.